Publish to Maven Central

jeggy opened the pull request:
closes #45

Tasks:
* [x] Setup SonaType account and get `com.apurebase` approved there
* [x] Gradle configuration for publishing to Maven Central
* [x] Make sure it's using 1.8
* [x] Setup Jupiter again
* [x] Create GitHub action to publish to Maven Central
* [x] Publish old versions to Maven Central

jeggy committed:
Converted to Kotlin Gradle Script & setup for maven central

AndreasVolkmann commented on arkenv-yaml/build.gradle.kts:
```diff
@@ -0,0 +1,91 @@
+
+plugins {
+    base
+    kotlin("jvm") version "1.3.72"
```
Any reason for lower version?

jeggy replied:
I just upgraded to the latest 1.3.x version. We can upgrade to 1.4 in some other branch 😄

AndreasVolkmann commented:
Nice to see `.kts` build files 👍

AndreasVolkmann commented:
The errors come from `Cannot inline bytecode built with JVM target 1.8 into bytecode that is being built with JVM target 1.6.`

We need something like this: 
```kotlin
compileOptions {
    sourceCompatibility = 1.8
    targetCompatibility = 1.8
}
tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
    kotlinOptions {
        jvmTarget = "1.8"
    }
}
```

jeggy commented:
This is still in `WIP` 😄 

But yeah, current things I want to fix is:
* Make sure it's using 1.8
* Setup Jupiter again
* Create GitHub action to publish to SonaType

jeggy committed:
Created github publish to maven action

jeggy committed:
Fixed sonatype credentials from env

jeggy committed:
Publish on github release

jeggy committed:
Run unit test github action

jeggy committed:
Updated Tests github action configuration

jeggy committed:
Updated docs to use mavenCentral instead of jcenter

jeggy commented on arkenv-yaml/build.gradle.kts:
```diff
@@ -0,0 +1,100 @@
+    compile(project(":arkenv"))
+    compile("org.yaml:snakeyaml:$snakeyamlVersion")
+
+    // Can't figure out how to extend via arkenv project test sourcesets
```
TODO: Forgot to remove this comment :smile:

jeggy committed:
Fix test

jeggy committed:
Merged master & fixed tests setup

jeggy committed:
Cleanup after moving to testFixtures

jeggy commented:
There I finally got Kotlin Gradle Script to fully play along :laughing: 

As some test classes were shared between `:arkenv` and `:arkenv-yaml` I have converted this into a testFixture project and moved all shared utilities into `arkenv/src/testFixtures` directory :+1: 

After this has been merged, releases will be automatically published to MavenCentral when a github release has been made, which can be see in `.github/workflows/maven-publish.yml`
